File: Java/PG/src/K번째수.java
import java.util.Arrays;

public class K번째수 {
    public static int[] solution(int [] array, int [][] commands){
        int [] answer = new int[commands.length];
        int i = 0;
        for( int [] command : commands){
            int start = command[0]-1, end = command[1], target=command[2];
            int [] temps = Arrays.copyOfRange(array, start, end);
            Arrays.sort(temps);
            answer[i] = temps[target-1];
            i+=1;
        }
        return answer;
    }
    public static void main(String[] args) {
        int [] answer = solution(new int [] {1,5,2,6,3,7,4},new int [][] {{2,5,3},{4,4,1},{1,7,3}});
        for( int i : answer){
            System.out.println(String.valueOf(i));
        }
    }
}

File: Test/K번째수.js
console.log(solution([1, 5, 2, 6, 3, 7, 4],[[2, 5, 3], [4, 4, 1], [1, 7, 3]]));
function solution(array, commands){
    let answer = [];
    for(const [start, end, idx] of commands){
        let target = array.slice(start-1, end).sort((a, b) => a - b);
        answer.push(target[idx-1]);
    }
    return answer;
}